pkg/diagnostics/workflow_metrics.go

```diff
@@ -0,0 +1,162 @@
+	"go.opencensus.io/tag"
+)
+
+var (
+	elapsedKey = tag.MustNewKey("elapsed")
+
+	reminderTypeKey = tag.MustNewKey("reminder_type")
+
+	executionTypeKey = tag.MustNewKey("execution_type")
+
+	isRetryableKey = tag.MustNewKey("is_retryable")
+)
+
+type workflowMetrics struct {
+	// Total successful workflow operation requests
+	workflowOperationsTotal *stats.Int64Measure
+	// Total failed workflow operation requests
+	workflowOperationsFailedTotal *stats.Int64Measure
+	// Workflow operation request's Response latency
+	workflowOperationsLatency *stats.Float64Measure
+
+	// Total workflow/activity reminders created
+	workflowRemindersTotal *stats.Int64Measure
+
+	// Total successful workflow/activity executions
+	workflowExecutionTotal *stats.Int64Measure
+	// Total failed workflow/activity executions
+	workflowExecutionFailedTotal *stats.Int64Measure
+	// Time taken to run a workflow to completion
+	workflowExecutionLatency *stats.Float64Measure
+
+	// Latency between execution request and actual execution
+	workflowSchedulingLatency *stats.Float64Measure
+
+	appID   string
+	enabled bool
+}
+
+func newWorkflowMetrics() *workflowMetrics {
+	return &workflowMetrics{
+		workflowOperationsTotal: stats.Int64(
+			"runtime/workflow/operations/total",
+			"The number of successful workflow operation requests.",
+			stats.UnitDimensionless),
+
+		workflowOperationsFailedTotal: stats.Int64(
+			"runtime/workflow/operations/failed_total",
+			"The number of failed workflow operations requests.",
+			stats.UnitDimensionless),
+		workflowOperationsLatency: stats.Float64(
+			"runtime/workflow/operations/latency",
+			"The latencies of responses for workflow operation requests.",
+			stats.UnitMilliseconds),
+		workflowRemindersTotal: stats.Int64(
+			"runtime/workflows/reminders/total",
+			"The number of workflows/activity reminders created.",
+			stats.UnitDimensionless),
+		workflowExecutionTotal: stats.Int64(
+			"runtime/workflow/execution/total",
+			"The number of successful workflow/activity executions.",
+			stats.UnitDimensionless),
+		workflowExecutionFailedTotal: stats.Int64(
+			"runtime/workflow/execution/failed_total",
+			"The number of failed workflow/activity executions.",
+			stats.UnitDimensionless),
+		workflowExecutionLatency: stats.Float64(
+			"runtime/workflow/execution/latency",
+			"The total time taken to run a workflow/activity to completion.",
+			stats.UnitMilliseconds),
+		workflowSchedulingLatency: stats.Float64(
+			"runtime/workflow/scheduling/latency",
+			"The latency between execution request and actual execution.",
+			stats.UnitMilliseconds),
+	}
+}
+
+func (w *workflowMetrics) IsEnabled() bool {
+	return w != nil && w.enabled
+}
+
+// Init registers the workflow metrics views.
+func (w *workflowMetrics) Init(appID string) error {
+	w.appID = appID
+	w.enabled = true
+
+	return view.Register(
+		diagUtils.NewMeasureView(w.workflowOperationsTotal, []tag.Key{appIDKey, operationKey}, view.Count()),
```

Comment:
Add namespace Key as well similar to other component metrics. Add a componentKey which would point to which component is being used(`dapr` for now). 

We also have a proposal to support different backends in Dapr(#7127). So adding a `Backendkey` would be useful as well(Currently the only supported value would be `ActorBackend`, but more would come later). 

And like said above a `Success` Key would help combining 2 metrics together

Reply:
Understood, also what would be the values for `namespace keys` ?

Reply:
The namespace can be added while initializing the metric. See this for example: https://github.com/dapr/dapr/blob/5050df5bf10835e9890b6b6519f5fa22e4736b28/pkg/diagnostics/metrics.go#L61

Reply:
@DeepanshuA wdyt? Since workflow actors have the namespace in their name, I believe adding it might be useful. It can be added to all the workflow metrics ideally